### PR TITLE
Nexus 22157 Handle component name difference between feature and plugin

### PR DESCRIPTION
**Pull request by proteus1121**
https://issues.sonatype.org/browse/NEXUS-22157

### CODE REVIEW

File: nexus-repository-p2/src/main/java/org/sonatype/nexus/repository/p2/internal/util/P2PathUtils.java [hunk 1]
```diff
@@ -123,7 +129,18 @@ public String filename(final TokenMatcher.State state) {
    * @see #name(State)
    */
   public String componentName(final TokenMatcher.State state) {
```
Comment (sergeyonofreychuck) [inline nexus-repository-p2/src/main/java/org/sonatype/nexus/repository/p2/internal/util/P2PathUtils.java]: add unit tests

File: nexus-repository-p2/src/main/java/org/sonatype/nexus/repository/p2/internal/proxy/P2ProxyFacetImpl.java [hunk 2]
```diff
@@ -353,7 +353,7 @@ protected P2Attributes mergeAttributesFromTempBlob(final TempBlob tempBlob, fina
     Optional<P2Attributes> p2Attributes = Optional.empty();
     // first try Features XML
     try (JarInputStream jis = getJar(tempBlob, sourceP2Attributes.getExtension())) {
-      p2Attributes = jarParser.getAttributesFromFeatureXML(jis);
+      p2Attributes = jarParser.getAttributesFromFeatureXML(jis, p2PathUtils);
```
Comment (sergeyonofreychuck) [inline nexus-repository-p2/src/main/java/org/sonatype/nexus/repository/p2/internal/proxy/P2ProxyFacetImpl.java]: convert p2PathUtils to static

File: nexus-repository-p2/src/main/java/org/sonatype/nexus/repository/p2/internal/util/P2PathUtils.java (hunk 1 above)
Comment (proteus1121) [reply nexus-repository-p2/src/main/java/org/sonatype/nexus/repository/p2/internal/util/P2PathUtils.java]: Done

File: nexus-repository-p2/src/main/java/org/sonatype/nexus/repository/p2/internal/proxy/P2ProxyFacetImpl.java (hunk 2 above)
Comment (proteus1121) [reply nexus-repository-p2/src/main/java/org/sonatype/nexus/repository/p2/internal/proxy/P2ProxyFacetImpl.java]: Done

File: nexus-repository-p2/src/main/java/org/sonatype/nexus/repository/p2/internal/util/P2PathUtils.java [hunk 3]
```diff
@@ -20,6 +20,8 @@
 import org.sonatype.nexus.repository.view.matchers.token.TokenMatcher;
```
Comment (sergeyonofreychuck) [inline nexus-repository-p2/src/main/java/org/sonatype/nexus/repository/p2/internal/util/P2PathUtils.java]: remove unused imports

Comment (proteus1121) [reply nexus-repository-p2/src/main/java/org/sonatype/nexus/repository/p2/internal/util/P2PathUtils.java]: Done

File: nexus-repository-p2/src/test/java/org/sonatype/nexus/repository/p2/internal/util/P2PathUtilsTest.java [hunk 4]
```diff
@@ -46,33 +43,32 @@
...
 
-  @Before
-  public void setUp() throws Exception {
-    p2PathUtils = new P2PathUtils();
-  }
+  private final String fakeComponentName = "eclipsepackage1";
+
+  private final String fakeFileName = fakeComponentName + "_2_3.jar";
+
+  private final String fakeExtension = "pack.gz";
```
Comment (sergeyonofreychuck) [inline nexus-repository-p2/src/test/java/org/sonatype/nexus/repository/p2/internal/util/P2PathUtilsTest.java]: use static imports

Comment (proteus1121) [reply nexus-repository-p2/src/test/java/org/sonatype/nexus/repository/p2/internal/util/P2PathUtilsTest.java]: Done

**Pull request merged**